source/utilities.cc

```diff
@@ -183,6 +183,54 @@ namespace aspect
     // Copyright (C) 2011, 2014 Tino Kluge (ttk448 at gmail.com)
     namespace tk
     {
+      /**
+       * Band matrix solver for a banded, square matrix of size @p dim. Used by spline interpolation in tk::spline.
+       */
+      class band_matrix
+      {
+        public:
+          band_matrix() {}                              // constructor
+          band_matrix(int dim, int n_u, int n_l);       // constructor
+          void resize(int dim, int n_u, int n_l);       // init with dim,n_u,n_l
```

Comment:
Would you mind bringing this in our usual style with doxygen formatting? I know it is just copy-paste on your part, but someone should do it, and right now I can force you to it :smile: